File: app/src/main/java/org/tasks/location/MapFragment.java
package org.tasks.location;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import java.util.List;
import org.tasks.data.Place;

public interface MapFragment {

  void init(AppCompatActivity activity, MapFragmentCallback callback, boolean dark);

  @Nullable MapPosition getMapPosition();

  void movePosition(MapPosition mapPosition, boolean animate);

  void setMarkers(List<Place> places);

  void disableGestures();

  void showMyLocation();

  void onPause();

  void onResume();

  void onDestroy();

  interface MapFragmentCallback {
    void onMapReady(MapFragment mapFragment);

    void onPlaceSelected(Place place);
  }
}

File: app/src/main/java/org/tasks/location/MapFragment.kt
package org.tasks.location

import androidx.appcompat.app.AppCompatActivity
import org.tasks.data.entity.Place

interface MapFragment {
    fun init(activity: AppCompatActivity, callback: MapFragmentCallback, dark: Boolean)

    val mapPosition: MapPosition?

    fun movePosition(mapPosition: MapPosition, animate: Boolean)

    fun setMarkers(places: List<Place>)

    fun disableGestures()

    fun showMyLocation()

    fun showCircle(radius: Double, latitude: Double, longitude: Double)

    fun onPause()

    fun onResume()

    fun onDestroy()

    interface MapFragmentCallback {
        fun onMapReady(mapFragment: MapFragment)
        fun onPlaceSelected(place: Place)
    }
}